scripts/train_agent_babyai.py

```diff
@@ -1,52 +1,88 @@
 import os
 
-import numpy as np
 import torch
 from transformers import DecisionTransformerConfig
 
 from src.utils.utils import log
 from src.utils.argparsing import get_args
 from src.dataset.custom_dataset import CustomDataset
-from src.collator import Collator
+from src.collator import Collator, RoundRobinCollator, CurriculumCollator
 from src.agent.fdt import MinigridFDTAgent
 from src.trainer import AgentTrainer
+from src.dataset.seeds import LEVELS_CONFIGS
+from src.constants import ENV_METADATA_PATH, OUTPUT_PATH
 
 os.environ["WANDB_DISABLED"] = "true"
-os.environ["ENV_METADATA_PATH"] = "/home/s2227283/projects/feedback-DT/env_metadata.jsonc"
+os.environ["ENV_METADATA_PATH"] = ENV_METADATA_PATH
 
 args = get_args()
 
-args["output"] = "/home/s2227283/projects/feedback-DT/data/output"
-args["run_name"] = "july-24-test-1"
-args["num_episodes"] = 500
-args["load_dataset_if_exists"] = False
-args["rgb_obs"] = True
-args["fully_obs"] = False
+args["output"] = OUTPUT_PATH
+args["run_name"] = "22-aug-test-final"
+# args["level"] = "GoToRedBallGrey"
+args["num_episodes"] = 20
 args["seed"] = 0
-args["policy"] = "random_used_action_space_only"
+args["policy"] = "random"
 args["wandb_mode"] = "disabled"
 args["report_to"] = "none"
 args["epochs"] = 5
 args["log_interval"] = 1
+args["train_mode"] = "anti_curriculum"
 
 frame_size = 64 if args["fully_obs"] else 56
+custom_order = [0, 1, 2, 4, 3, 5, 9, 8, 7, 6, 10, 15, 16, 14, 13, 18, 17]
 
 log("setting up devices")
 if torch.cuda.is_available():
     device = torch.device("cuda")
     device_str = f"{device.type}:{device.index}" if device.index else f"{device.type}"
     os.environ["CUDA_VISIBLE_DEVICES"] = device_str
-    # log(f"Using device: {torch.cuda.get_device_name()}")
+    log(f"Using device: {torch.cuda.get_device_name()}")
     log("using gpu")
+if not torch.backends.mps.is_available():
+    if not torch.backends.mps.is_built():
+        print(
+            "MPS not available because the current PyTorch install was not "
+            "built with MPS enabled."
+        )
+    else:
+        print(
+            "MPS not available because the current MacOS version is not 12.3+ "
+            "and/or you do not have an MPS-enabled device on this machine."
+        )
+        device = torch.device("cpu")
+        log("using cpu")
 else:
-    device = torch.device("cpu")
-    log("using cpu")
+    device = torch.device("mps")
+    log("using mps")
 
-log("creating dataset...")
-dataset = CustomDataset.get_dataset(args)
+if not "single" in args["train_mode"]:
```

Comment:
would be great to document somewhere what values are allowed for `train_mode` (apologies if you already did and i missed it)

Reply:
Let me know if that's not enough and you want it somewhere else too